packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentEditor.tsx

```diff
@@ -149,7 +149,13 @@ export default function ComponentEditor({ className }: ComponentEditorProps) {
 
   const { selection } = editor;
 
-  const selectedNode = selection ? appDom.getNode(dom, selection) : null;
+  let selectedNode;
+
+  try {
+    selectedNode = selection ? appDom.getNode(dom, selection) : null;
+  } catch (error) {
```

Comment:
I'm not really sure if this is the right way to deal with it. I bumped into this issue when I add item, it becomes selected, then upon undo it gets removed, but selection ref remains.

IMO we might want to consider to include selected state in the dom, I'm seeing it's pretty common in other apps to also switch selection state with undo/redo and I think it's fairly good UX

Reply:
use `appDom.getMaybeNode`. If the selected node should always exist, otherwise it means there's a programmer error, then use an `invariant` and just let it crash.

Reply:
Thanks, I guess that's what I was missing here, I will use `getMaybeNode`